### DISCUSSION TITLE
issue modifying LOS services.jar

### DISCUSSION THREAD

**Discussion by farfromrefug** [iBotPeaches/Apktool#3653] [Q&A] [answered]
I have a script which modifies services.jar taken from LOS rom.
Just decoding/building again the services.jar creates error after LOS boot with that new services.jar.
I tried many different parameters to apktool, also tried latest build from repo, and still always the same.
I end up with errors like this:
```
java.lang.String com.android.server.policy.SingleKeyGestureDetector$MessageObject.toString() failed to verify: java.lang.String com.android.server.policy.SingleKeyGestureDetector$MessageObject.toString(): [0x0] invalid argument count (1) exceeds outsSize (0) (declaration of 'com.android.server.policy.SingleKeyGestureDetector$MessageObject' appears in /system/framework/services.jar!classes2.dex)
                         E      at com.android.server.policy.SingleKeyGestureDetector.interceptKeyDown(SingleKeyGestureDetector.java:286)
                         E      at com.android.server.policy.SingleKeyGestureDetector.interceptKey(SingleKeyGestureDetector.java:220)
                         E      at com.android.server.policy.PhoneWindowManager.handleKeyGesture(PhoneWindowManager.java:5795)
                         E      at com.android.server.policy.PhoneWindowManager.interceptKeyBeforeQueueing(PhoneWindowManager.java:5295)
                         E      at com.android.server.wm.InputManagerCallback.interceptKeyBeforeQueueing(InputManagerCallback.java:165)
                         E      at com.android.server.input.InputManagerService.interceptKeyBeforeQueueing(InputManagerService.java:2563)
```
It seems like the method `com.android.server.policy.SingleKeyGestureDetector$MessageObject.toString()` does not have the same signature in the rebuilt services.jar. Would someone have any idea on what the issue could be?

I use something like this:
```
apktool d -r -f d/system/framework/services.jar -o services_temp
apktool b -c -f services_temp -o services.jar
```

**Comment by iBotPeaches** [comment, marked as answer]
You might have some luck reporting upstream in smali. I see an issue that looks close to this - https://github.com/google/smali/issues/60

**Reply by farfromrefug** [reply]
OK thanks. Will try to report it. Seems like everyone is facing that issue now on android 14. 
I just found that forum posts https://xdaforums.com/t/module-smalipatcherex-1-1-2.4627905/page-4 where they describe exactly the same power button error (this is when i can the error described here).
